lib/screens/redux/reducers.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:chat/redux/states.dart';
+import 'package:flutter/material.dart';
+import 'actions.dart';
+
+ScreenMode appReducers(ScreenMode previousScreenMode, dynamic action) {
+  if (action is ChangeScreenMode && !action.isDark) {
```

Comment:
1. inside reducer, it's better to not mix checking action type & payload: first check action type, and then inside action handler check the payload
2. here it can be easily written as 
```
if (action is ChangeScreenMode) {
    const color = action.isDark? xxx : yyy;
    return const ScreenMode(color, action.isDark);
}
```